kernel/src/engine/arrow_conversion.rs

```diff
@@ -206,11 +206,19 @@ impl TryFrom<&ArrowDataType> for DataType {
             ArrowDataType::Binary => Ok(DataType::Primitive(PrimitiveType::Binary)),
             ArrowDataType::FixedSizeBinary(_) => Ok(DataType::Primitive(PrimitiveType::Binary)),
             ArrowDataType::LargeBinary => Ok(DataType::Primitive(PrimitiveType::Binary)),
-            ArrowDataType::Decimal128(p, s) => Ok(DataType::decimal(*p, *s)),
+            ArrowDataType::Decimal128(p, s) => {
+                if p > &38 || s > &38 {
```

Comment:
As far as I know Arrow Decimal128 can't hold larger than 38 precision/scale, so this check seems unnecessary

Reply:
the input can hold `i8`, `u8` values, which of course would be wrong, but users could construct it. since we a re fallible here, I though why not check ...

in other cases where we are currently not on a fallible code path I omitted the checks thus far, where there may be more need for it. for instance the `DataType::decimal` function below does not check internally so far, but I'll check how much that bubbles up ...